src/Altinn.App.Core/Models/Layout/BaseComponent.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Immutable;
+using System.Text.Json;
+
+using Altinn.App.Core.Models.Expressions;
+
+namespace Altinn.App.Core.Models.Layout;
+
+
+/// <summary>
+/// Inteface to be able to handle all most components same way.
+/// </summary>
+/// <remarks>
+/// See <see cref="GroupComponent" /> for any components that handle children.
+/// Includes <see cref="DataModelBindings" /> that will be initialized to an empty dictionary
+/// for components that don't have them.
+/// </remarks>
+public class BaseComponent
+{
+    /// <summary>
+    /// Constructor that ensures n
+    /// </summary>
+    public BaseComponent(string id, string type, IReadOnlyDictionary<string, string>? dataModelBindings, Expression? hidden, Expression? required, Expression? readOnly, IReadOnlyDictionary<string, string>? extra)
+    {
+        Id = id;
+        Type = type;
+        DataModelBindings = dataModelBindings ?? ImmutableDictionary<string, string>.Empty;
+        Hidden = hidden;
+        Required = required;
+        ReadOnly = readOnly;
+        Extra = extra;
+    }
+    /// <summary>
+    /// ID of the component (or pagename for pages)
+    /// </summary>
+    public string Id { get; }
+
+    /// <summary>
+    /// Get the page for the component
+    /// </summary>
+    public string Page
```

Comment:
Page | PageId | PageName 🤔 Page feels like it's a Page object, but it's just a id/name as a string. Might be a review-in-browser-thing and easier to reason about in and IDE.

Reply:
Haha! Great catch! I got confused about this today🫣🫣.

Reply:
[b731c8c](https://github.com/Altinn/app-lib-dotnet/pull/23/commits/b731c8c85d54b5136b2639f60f7e1738be6595b7)